src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt

```diff
@@ -0,0 +1,108 @@
+package chip.setuppayload
+
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/** Parser for scanned QR code or manual entry code.  */
+class SetupPayloadParser {
+  /**
+   * Returns [SetupPayload] parsed from the QR code string. If an invalid element is included
+   * in the QRCode Parse result, SetupPayloadException occurs. Refer to [SetupPayload] for the
+   * description of the invalid element.
+   */
+  @Throws(UnrecognizedQrCodeException::class, SetupPayloadException::class)
+  fun parseQrCode(qrCodeString: String): SetupPayload {
+    return fetchPayloadFromQrCode(qrCodeString, false)
+  }
+
+  /**
+   * Returns [SetupPayload] parsed from the QR code string.
+   *
+   * @param qrCodeString the QRCode for commissioning device.
+   * @param allowInvalidPayload Allow invalid payload components. If this value is true, payload
```

Comment:
Why would we allow invalid payloads?

Reply:
We are porting existing code, not re-making it.

#22357 seems to have added it, I am guessing it is for some tests that are run and you must accept invalid data somehow. Changing tests is outside of the scope of a port. Possible and desirable I believe, however as a separate task (and I would suggest after things are moved to use less C++).

Reply:
The name is probably bad .... skipContentValidation is probably a better name based on the comments in the original PR.

Reply:
Renamed to skipPayloadValidation